python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -156,26 +316,78 @@ def cache_session_data(host, login, session_token):
-    info_path = _get_session_data_location(host)
-
-    # make sure the info_dir exists!
-    info_dir, info_file = os.path.split(info_path)
-    if not os.path.exists(info_dir):
-        os.makedirs(info_dir, 0700)
-
-    logger.debug("Caching login info at %s...", info_path)
-    # Create a document with the following format:
-    # [LoginInfo]
-    # login=username
-    # session_token=some_unique_id
-    config = SafeConfigParser()
-    config.add_section("LoginInfo")
-    config.set("LoginInfo", "login", login)
-    config.set("LoginInfo", "session_token", session_token)
-    # Write it to disk.
-    with open(info_path, "w") as configfile:
-        config.write(configfile)
-    logger.debug("Cached!")
+    file_path = _get_users_file_location(host)
+    _ensure_folder_for_file(file_path)
+
+    logger.debug("Caching login info at %s...", file_path)
+
+    document = _try_load_users_file(file_path)
+
+    if _insert_or_update_user(document, login, session_token):
+        # Write back the file only it a new user was added.
+        _write_yaml_file(file_path, document)
+        logger.debug("Cached!")
+    else:
+        logger.debug("Already cached!")
+
+
+def get_current_user(host):
+    """
+    Returns the current user for the given host.
+
+    :param host: Host to fetch the current for.
+
+    :returns: The current user for this host or None if not set.
+    """
+    # Retrieve the cached info file location from the host
+    info_path = _get_current_user_file_location(host)
+    if os.path.exists(info_path):
+        document = _try_load_current_user_file(info_path)
+        return document["current_user"]
+    return None
+
+
+def set_current_user(host, login):
+    """
+    Saves the current user for a given host.
+
+    :param host: Host to save the current user for.
+    :param login: The current user login for specified host.
+    """
+    file_path = _get_current_user_file_location(host)
+    _ensure_folder_for_file(file_path)
+
+    current_user_file = _try_load_current_user_file(file_path)
+    current_user_file["current_user"] = login
+    _write_yaml_file(file_path, current_user_file)
+
+
+def get_current_host():
+    """
+    Returns the current host.
+
+    :returns: The current host string.
+    """
+    # Retrieve the cached info file location from the host
+    info_path = _get_current_host_file_location()
+    if os.path.exists(info_path):
+        document = _try_load_current_host_file(info_path)
+        return document["current_host"]
+    return None
+
+
+def set_current_host(host):
+    """
+    Saves the current host.
+
+    :param host: The new current host.
+    """
+    file_path = _get_current_host_file_location()
+    _ensure_folder_for_file(file_path)
+
+    current_host_file = _try_load_current_host_file(file_path)
+    current_host_file["current_host"] = host
+    _write_yaml_file(file_path, current_host_file)
 
 
 def generate_session_token(hostname, login, password, http_proxy):
```

Comment:
I guess this method is different that the others, in that it doesn't have anything to do with the caching of things - it's more a utility thing - So I guess if we turn things into a class, this would be a separate piece!